Assets/UI/unitflagmanager_CQUI.lua

```diff
@@ -344,6 +350,73 @@ function OnUnitPromotionChanged( playerID : number, unitID : number )
     end
 end
 
+-- ===========================================================================
+--  CQUI modified OnLensLayerOn function
+--  If Religion Lens is enabled, call Refresh so non-religious units can be dimmed
+-- ===========================================================================
+function OnLensLayerOn( layerNum:number )
+    BASE_CQUI_OnLensLayerOn(layerNum);
+
+    if (layerNum == m_HexColoringReligion) then
+        m_IsReligionLensOn = true;
+        -- Call Refresh, which will cycle through all of the units visible to the current player
+        -- OnUnitVisibilityChanged will be called for every unit the player can see, where the
+        -- icon can be dimmed due to the religion lens being selected.
+        Refresh();
+    end
+end
+
+-- ===========================================================================
+--  CQUI modified OnLensLayerOff function
+--  If Religion Lens is disabled, call Refresh so non-religious units can be restored
+-- ===========================================================================
+function OnLensLayerOff( layerNum:number )
+    BASE_CQUI_OnLensLayerOff(layerNum);
+
+    if (layerNum == m_HexColoringReligion) then
+        m_IsReligionLensOn = false;
+        Refresh();
+    end
+end
+
+-- ===========================================================================
+--  CQUI modified OnUnitVisibilityChanged function
+--  Override basegame OnUnitVisibilityChanged to dim non-religious icons when the religion lens is displayed
+-- ===========================================================================
+function OnUnitVisibilityChanged( playerID: number, unitID : number, eVisibility : number )
+    -- print("***** OnUnitVisibilityChanged playerID:"..tostring(playerID).."  unitID:"..tostring(unitID).."  evis:"..tostring(eVisibility));
+    local flagInstance = GetUnitFlag( playerID, unitID );
+    if (flagInstance ~= nil) then
+        flagInstance:SetFogState( eVisibility );
+        flagInstance:UpdatePosition();
+
+        -- ===== CQUI Changes: Dim non-religion units if religious lens is on =====
+        if m_IsReligionLensOn then
+            if (flagInstance.m_Style ~= FLAGSTYLE_RELIGION) then
+                -- FLAGSTYLE_RELIGION is a Global constant in UnitFlagManager.lua
+                -- Dim the flag and health for non-religious units; if dimmed already (no moves, sleeping), dim more to distinguish
+                local alphaValue = 0.65;
```

Comment:
Since this is the same as a "default" dim, isn't it somwhere defined to we can use it and not hardcode?